blog/2025-02-21/index.md

```diff
@@ -0,0 +1,123 @@
+
+We mirror the exact sources each app is built against in case the original source goes down or there is some other issue, and anyone can build the Flatpak back from those mirrored source to reproduce or audit the build. The manifest used to build the app is distributed to every user in the app’s sandbox at `/app/manifest.json`, which can be inspected and used to rebuild the app exactly as it was built by Flathub.
+
+## Verification
+
+Apps can be [verified](https://docs.flathub.org/docs/for-app-authors/verification) on Flathub; this process confirms that an app is published by the original developer or an authorized party by proving ownership of the app ID. While all apps are held to the same high standards of safety and review on Flathub, this extra layer helps users confirm that the app they are getting is _also_ provided or authorized by its developer.
+
+![Verified checkmark](verified.png)
+
+Over half of the apps on Flathub so far are verified, with the number regularly increasing.
+
+## App Store Clients
+
+Once an app is developed, submitted, tested, approved, built, and distributed, it appears in app store clients like Flathub.org, KDE Discover, GNOME Software, and elementary AppCenter—as well as the Flatpak CLI. While exact implementations vary and the presentation is up to the specific app store client, generally each will show:
+
+- Static permissions and their impact on safety
+- Open Age Rating Service rating and details
+- If an app uses outdated runtimes
+- Release notes for each release
+- If static permissions increase between releases
+
+Flathub.org and GNOME Software also display the app’s verified status.
+
+## Updates
+
+Once an app is accepted onto Flathub, it’s not just the wild west; there are still a number of safety protections built into the flow:
+
+- **Flathub maintains ownership over the manifest repo**, while app developers are invited as limited collaborators
+- **The manifest’s default branch is protected**, preventing direct pushes without a pull request
+- **Flathub’s automated tests must pass** before a PR can be merged and an update can be pushed
+- **Static permission changes are held for human review** before an update is released to users
+- **Critical MetaInfo changes are held for human review**, e.g. if an app changes its name or license
+
+## Caveats
+
+There are a few caveats to some of the points above which I would be remiss not to mention. First, Flathub has granted a small handful of trusted partners (including Mozilla and OBS Studio) the ability to directly upload their builds from their own infrastructure. Even for the few cases of direct uploads, we require a public manifest and build pipeline to enable similar reproducibility and auditability as outlined above. We also require the apps to be verified, and still run automated tests such as our linter against them.
```

Comment:
When I raised this point the other day, people thought that applications were published "untested" even though these projects have an entire CI pipeline which validates the state of the application and have done QA work before tagging the release.

Reply:
Hm, interesting. I'll try to clear that out

Reply:
What about:

```suggestion
There are a few caveats to some of the points above which I would be remiss not to mention.

First, Flathub has granted a small handful of trusted partners (including Mozilla and OBS Studio) the ability to directly upload their builds from their own infrastructure. These projects have an entire CI pipeline which validates the state of their app, and they perform QA before tagging the release and pushing it to Flathub. Even for these few cases of direct uploads, we require a public manifest and build pipeline to enable similar reproducibility and auditability as outlined above. We also require the apps to be verified, and still run automated tests such as our linter against them.
```